### PR TITLE
docs: Make it clear that harvestRPDE runs indefinitely except for fatal error

### LINKED ISSUE #7: Document returning behaviour of harvestRPDE
Make it clear that harvestRPDE runs indefinitely and only returns if there is a fatal error

### PR DESCRIPTION
**Pull request by lukehesluke**
closes #7

### CODE REVIEW

File: src/harvest-rpde.js [hunk 1]
```diff
@@ -200,6 +203,7 @@ async function harvestRPDE({
         return;
       }
       if (error.response?.status === 404) {
+        // TODO document: why?
```
Comment (nickevansuk) [inline src/harvest-rpde.js:206]:
```suggestion
        // As per https://openactive.io/realtime-paged-data-exchange/#http-status-codes, consider this endpoint in an error state and do not retry
```

File: src/harvest-rpde.js [hunk 2]
```diff
@@ -200,6 +203,7 @@ async function harvestRPDE({
         return;
       }
       if (error.response?.status === 404) {
```
Comment (nickevansuk) [inline src/harvest-rpde.js:205]: Perhaps for consistency with the spec?

```suggestion
      if (error.response?.status === 404 || error.response?.status === 410) {
```

File: src/harvest-rpde.js [hunk 3]
```diff
@@ -183,6 +185,7 @@ async function harvestRPDE({
       if ((WAIT_FOR_HARVEST || VALIDATE_ONLY) && isOrdersFeed) {
         onFeedEnd();
       }
+      // TODO document: What is a 'FatalError'?
```
Comment (nickevansuk) [inline src/harvest-rpde.js:188]: Interestingly we've lost the definition of FatalError here, where it was previously slightly more well-defined:

https://github.com/openactive/openactive-test-suite/blob/d81376347a3419287ba0ee05b7fb399322c09485/packages/openactive-broker-microservice/app.js#L300

**Review by nickevansuk — APPROVED** [review]
Approved with some suggestions

File: src/harvest-rpde.js (hunk 3 above)
Comment (civsiv) [reply src/harvest-rpde.js:188]: "FatalError" seems very broker-specific though and probably shouldn't be in the library? Is it work unpicking this further?

**Review by civsiv — APPROVED** [review]

File: src/harvest-rpde.js (hunk 3 above)
Comment (lukehesluke) [reply src/harvest-rpde.js]: I now understand this, thanks! I've updated the comment to explain this and suggest decoupling it in future and have created this issue to house all these decoupling-related tasks: https://github.com/openactive/harvesting-utils/issues/9

**Pull request merged**